pyomo/gdp/tests/test_binary_multiplication.py

```diff
@@ -0,0 +1,316 @@
+        ct.check_xor_constraint(self, 'binary_multiplication')
+
+    def test_or_constraints(self):
+        m = models.makeTwoTermDisj()
+        m.disjunction.xor = False
+        TransformationFactory('gdp.binary_multiplication').apply_to(m)
+
+        # check or constraint is an or (upper bound is None)
+        orcons = m._pyomo_gdp_binary_multiplication_reformulation.component(
+            "disjunction_xor"
+        )
+        self.assertIsInstance(orcons, Constraint)
+        assertExpressionsEqual(
+            self,
+            orcons.body,
+            EXPR.LinearExpression(
+                [
+                    EXPR.MonomialTermExpression((1, m.d[0].binary_indicator_var)),
+                    EXPR.MonomialTermExpression((1, m.d[1].binary_indicator_var)),
+                ]
+            ),
+        )
+        self.assertEqual(orcons.lower, 1)
+        self.assertIsNone(orcons.upper)
+
+    def test_deactivated_constraints(self):
+        ct.check_deactivated_constraints(self, 'binary_multiplication')
+
+    def test_transformed_constraints(self):
+        m = models.makeTwoTermDisj()
+        binary_multiplication = TransformationFactory('gdp.binary_multiplication')
+        binary_multiplication.apply_to(m)
+        self.check_transformed_constraints(m, binary_multiplication, -3, 2, 7, 2)
+
+    def test_do_not_transform_userDeactivated_disjuncts(self):
+        ct.check_user_deactivated_disjuncts(self, 'binary_multiplication')
+
+    def test_improperly_deactivated_disjuncts(self):
+        ct.check_improperly_deactivated_disjuncts(self, 'binary_multiplication')
+
+    def test_do_not_transform_userDeactivated_IndexedDisjunction(self):
+        ct.check_do_not_transform_userDeactivated_indexedDisjunction(
+            self, 'binary_multiplication'
+        )
+
+    # helper method to check the M values in all of the transformed
+    # constraints (m, M) is the tuple for M.  This also relies on the
+    # disjuncts being transformed in the same order every time.
+    def check_transformed_constraints(
+        self, model, binary_multiplication, cons1lb, cons2lb, cons2ub, cons3ub
+    ):
+        disjBlock = (
+            model._pyomo_gdp_binary_multiplication_reformulation.relaxedDisjuncts
+        )
+
+        # first constraint
+        c = binary_multiplication.get_transformed_constraints(model.d[0].c)
+        self.assertEqual(len(c), 1)
+        c_lb = c[0]
+        self.assertTrue(c[0].active)
+        repn = generate_standard_repn(c[0].body)
+        self.assertIsNone(repn.nonlinear_expr)
+        self.assertEqual(len(repn.quadratic_coefs), 1)
+        self.assertEqual(len(repn.linear_coefs), 1)
+        ind_var = model.d[0].indicator_var
+        ct.check_quadratic_coef(self, repn, model.a, ind_var, 1)
+        ct.check_linear_coef(self, repn, ind_var, -model.d[0].c.lower)
+        self.assertEqual(repn.constant, 0)
+        self.assertEqual(c[0].lower, 0)
+        self.assertIsNone(c[0].upper)
```

Comment:
You don't have to change this if you don't have time, but I've been trying to move to using `assertExpressionsEqual` instead of this mess with the standard repn because it makes for much more readable tests.

Reply:
It was easy to change and is indeed far more readable.